src/hffs/fs.py

```diff
@@ -195,18 +208,21 @@ def rm(self, path, recursive=False, maxdepth=None, **kwargs):
             repo_type=repo_type,
             token=self.token,
             operations=operations,
-            revision=self.revision,
+            revision=revision,
             commit_message=kwargs.get("commit_message", commit_message),
             commit_description=kwargs.get("commit_description"),
         )
-        self.invalidate_cache(path=path)
+        self.invalidate_cache(path=self._unresolve_path(repo_type, repo_id, revision, path_in_repo))
 
-    def ls(self, path, detail=True, refresh=False, **kwargs):
+    def ls(self, path, detail=True, refresh=False, revision: Optional[str] = None, **kwargs):
         path = self._strip_protocol(path)
+        repo_type, repo_id, revision_in_path, path_in_repo = self.resolve_path(path)
+        repo_type, repo_id, revision, path_in_repo = self._resolve_path_with_revision(path, revision)
```

Comment:
Here it feels weird to parse twice the path. Is it because you need the initial `revision_in_path` no matter what? If that's the case, why not.

In general, do you think it would be possible to merge `resolve_path` and `_resolve_path_with_revision` together? Something like:

```py
def resolve_path(self, path: str, revision: Optional[str]) -> Tuple[str, str, Optional[str], str]:
```

or is this strictly forbidden by `fsspec` ? 

--- 
Note: if we merge both and return a ResolvedPath object, it could contain both `revision` and `revision_in_path` keys (and even `repo_type_in_path` for consistency when we unresolve the path?)